source/iface/command-accessor.rst

```diff
@@ -6,281 +6,274 @@
-    id<dimensions> index) const;
+.. _tags_buff_accessors:
 
-  *Single dimension in multi-dimensional buffer*
-  __unspecified__ &operator[](size_t index) const;
+=============================
+Buffer command accessors tags
+=============================
 
-Returns reference or value of element in the associated buffer at the
-requested index.
+Some ``accessor`` constructors take a ``TagT`` parameter,
+which is used to deduce template arguments.
+The permissible values for this parameter are listed in table below.
 
-One dimensional buffers are indexed by a data of type
-size_t. Multi-dimensional buffers may be indexed by a data of type
-``id<dimensions>``, or by a sequence of *[]*, 1 per dimension. For
-example ``a[1][2]``.  The operator returns a reference when the
-accessor allows writes, which requires that ``accessMode`` be one of
-``access::mode::write``, ``accessMode == access::mode::read_write``,
-``accessMode == access::mode::discard_write``, or ``accessMode ==
-access::mode::discard_read_write``. The operator returns an atomic if
-the ``accessMode`` is ``access::mode::atomic``.
+.. list-table::
+  :header-rows: 1
 
+  * - Tag value
+    - Access mode
+    - Accessor target
+  * - ``read_write;``
+    - ``access_mode::read_write``
+    - ``target::device``
+  * - ``read_only;``
+    - ``access_mode::read``
+    - ``target::device``
+  * - ``write_only;``
+    - ``access_mode::write``
+    - ``target::device``
+  * - ``read_write_host_task;``
+    - ``access_mode::read_write``
+    - ``target::host_task``
+  * - ``read_only_host_task;``
+    - ``access_mode::read``
+    - ``target::host_task``
+  * - ``write_only_host_task;``
+    - ``access_mode::write``
+    - ``target::host_task``
 
-``get_pointer``
-===============
 
-.. parsed-literal::
+``read-only accessors``
+=======================
 
-  *Available only when:
-   accessTarget == access::target::host_buffer*
+.. list-table::
+  :header-rows: 1
 
-  dataT \*get_pointer() const;
+  * - Data type
+    - Access mode
+  * - not const-qualified
+    - ``access_mode::read``
+  * - const-qualified
+    - ``access_mode::read``
 
-  *Available only when:
-   accessTarget == access::target::global_buffer*
+The specializations of ``accessor`` with
+``target::device or target::host_task`` that are read-only accessors.
+There is an implicit conversion between any of these specializations,
+provided that all other template parameters are the same.
 
-  sycl::global_ptr<dataT> get_pointer() const;
+``read-write accessors``
+========================
 
-  *Available only when:
-   accessTarget == access::target::constant_buffer*
+.. list-table::
+  :header-rows: 1
 
-  sycl::constant_ptr<dataT> get_pointer() const;
+  * - Data type
+    - Access mode
+  * - not const-qualified
+    - ``access_mode::read_write``
 
-Returns pointer to memory in a host buffer.
+The table present an implicit conversion from
+the read-write specialization.
 
+========================
+Common types and members
```

Comment:
Still believe that we should add common types here. Let's wait for Khronos for this decision

Reply:
Let's wait response from Khronos